// @tarojs/components/dist/esm-es5/taro-form-core.entry.js
import { r as registerInstance, c as createEvent, h, g as getElement } from './core-f216d924.js';
var Form = /** @class */ (function () {
    function Form(hostRef) {
        registerInstance(this, hostRef);
        this.value = {};
        this.onSubmit = createEvent(this, "submit", 7);
    }
    Form.prototype.watchSlotParent = function (newParent) {
        if (!this.orginalAppendChild) {
            this.orginalAppendChild = this.el.appendChild;
            this.orginalInsertBefore = this.el.insertBefore;
            this.orginalReplaceChild = this.el.replaceChild;
            this.orginalRemoveChild = this.el.removeChild;
        }
        if (!newParent) {
            this.el.appendChild = this.orginalAppendChild;
            this.el.insertBefore = this.orginalInsertBefore;
            this.el.replaceChild = this.orginalReplaceChild;
            this.el.removeChild = this.orginalRemoveChild;
            return;
        }
        this.el.appendChild = function (newChild) {
            return newParent.appendChild(newChild);
        };
        this.el.insertBefore = function (newChild, refChild) {
            return newParent.insertBefore(newChild, refChild);
        };
        this.el.replaceChild = function (newChild, oldChild) {
            return newParent.replaceChild(newChild, oldChild);
        };
        this.el.removeChild = function (oldChild) {
            return newParent.removeChild(oldChild);
        };
    };
    Form.prototype.onButtonSubmit = function (e) {
        e.stopPropagation();
        this.value = this.getFormValue();
        this.onSubmit.emit({
            value: this.value
        });
    };
    Form.prototype.onButtonReset = function (e) {
        e.stopPropagation();
        this.form.reset();
    };
    Form.prototype.componentDidLoad = function () {
        var _this = this;
        this.value = this.getFormValue();
        Object.defineProperty(this.el, 'value', {
            get: function () { return _this.value; },
            configurable: true
        });
    };
    Form.prototype.componentDidRender = function () {
        this.setSlotParent(this.form);
    };
    Form.prototype.getFormValue = function () {
        var el = this.el;
        var elements = [];
        var tagElements = el.getElementsByTagName('input');
        for (var j = 0; j < tagElements.length; j++) {
            elements.push(tagElements[j]);
        }
        var formItem = {};
        var hash = {};
        elements.forEach(function (item) {
            if (item.className.indexOf('weui-switch') !== -1) {
                formItem[item.name] = item.checked;
                return;
            }
            if (item.type === 'radio') {
                if (item.checked) {
                    hash[item.name] = true;
                    formItem[item.name] = item.value;
                }
                else {
                    if (!hash[item.name]) {
                        formItem[item.name] = '';
                    }
                }
                return;
            }
            if (item.type === 'checkbox') {
                if (item.checked) {
                    if (hash[item.name]) {
                        formItem[item.name].push(item.value);
                    }
                    else {
                        hash[item.name] = true;
                        formItem[item.name] = [item.value];
                    }
                }
                else {
                    if (!hash[item.name]) {
                        formItem[item.name] = [];
                    }
                }
                return;
            }
            formItem[item.name] = item.value;
        });
        var textareaElements = el.getElementsByTagName('textarea');
        var textareaEleArr = [];
        for (var i = 0; i < textareaElements.length; i++) {
            textareaEleArr.push(textareaElements[i]);
        }
        textareaEleArr.forEach(function (v) {
            formItem[v.name] = v.value;
        });
        return formItem;
    };
    Form.prototype.setSlotParent = function (el) {
        this.slotParent = el;
    };
    Form.prototype.render = function () {
        var _this = this;
        return (h("form", { ref: function (dom) {
                if (dom) {
                    _this.form = dom;
                }
            } }, h("slot", null)));
    };
    Object.defineProperty(Form.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Form, "watchers", {
        get: function () {
            return {
                "slotParent": ["watchSlotParent"]
            };
        },
        enumerable: true,
        configurable: true
    });
    return Form;
}());
export { Form as taro_form_core };
